test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelpersTest.cs

```diff
@@ -54,38 +54,62 @@ public class TagHelpersTests
+        public static TheoryData TagHelpersAreInheritedFromGlobalImportPagesData
         {
             get
             {
-                var expected1 =
-@"<root>root-content</root>
-
-
-<nested>nested-content</nested>";
-                yield return new[] { "NestedViewStartTagHelper", expected1 };
-
-                var expected2 =
-@"layout:<root>root-content</root>
-
-
-<nested>nested-content</nested>";
-
-                yield return new[] { "ViewWithLayoutAndNestedTagHelper", expected2 };
-
-                var expected3 =
-@"layout:<root>root-content</root>
-
-
-page:<root/>
-<nested>nested-content</nested>";
-                yield return new[] { "ViewWithInheritedRemoveTagHelper", expected3 };
+                // action, expected
+                return new TheoryData<string, string>
+                {
+                    {
+                        "NestedGlobalImportTagHelper",
+                        string.Format(
+                            "<root>root-content</root>{0}{0}{0}<nested>nested-content</nested>",
+                            Environment.NewLine)
+                    },
+                    {
+                        "ViewWithLayoutAndNestedTagHelper",
+                        string.Format(
+                            "layout:<root>root-content</root>{0}{0}{0}<nested>nested-content</nested>",
+                            Environment.NewLine)
+                    },
+                    {
+                        "ViewWithInheritedRemoveTagHelper",
+                        string.Format(
+                            "layout:<root>root-content</root>{0}{0}{0}page:<root/>{0}<nested>nested-content</nested>",
+                            Environment.NewLine)
+                    },
+                    {
+                        "ViewWithInheritedTagHelperPrefix",
+                        string.Format(
+                            "layout:<root>root-content</root>{0}{0}{0}page:<root>root-content</root>",
+                            Environment.NewLine)
+                    },
+                    {
+                        "ViewWithOverriddenTagHelperPrefix",
+                        string.Format(
+                            "layout:<root>root-content</root>{0}{0}{0}{0}page:<root>root-content</root>",
+                            Environment.NewLine)
+                    },
+                    {
+                        "ViewWithNestedInheritedTagHelperPrefix",
+                        string.Format(
+                            "layout:<root>root-content</root>{0}{0}{0}page:<root>root-content</root>",
+                            Environment.NewLine)
+                    },
+                    {
+                        "ViewWithNestedOverriddenTagHelperPrefix",
+                        string.Format(
+                            "layout:<root>root-content</root>{0}{0}{0}{0}page:<root>root-content</root>",
+                            Environment.NewLine)
+                    },
+                };
```

Comment:
Each functional test validates a specific aspect of the feature that's observable from a functional test standpoint only. Without these tests we'd have 0 coverage for end-to-end tests ensuring that `TagHelper`s get the proper prefixes and therefore output properly when being inherited from `_GlobalImport` files.